src/libs/ReportUtils.ts

```diff
@@ -3441,93 +3409,76 @@ function chatIncludesChronos(report) {
  * - It was written by someone else and isn't a whisper
  * - It's a welcome message whisper
  * - It's an ADDCOMMENT that is not an attachment
- *
- * @param {Object} reportAction
- * @param {number} reportID
- * @returns {Boolean}
  */
-function canFlagReportAction(reportAction, reportID) {
+function canFlagReportAction(reportAction: OnyxEntry<ReportAction>, reportID: string | undefined): boolean {
     const report = getReport(reportID);
-    const isCurrentUserAction = reportAction.actorAccountID === currentUserAccountID;
-
+    const isCurrentUserAction = reportAction?.actorAccountID === currentUserAccountID;
+    const isOriginalMessageHaveHtml =
+        reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT ||
+        reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.RENAMED ||
+        reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.CHRONOSOOOLIST;
```

Comment:
Why is this new condition added?

Reply:
only for those actionNames, originalMessage will have html, so its for type narrowing